src/wallet/wallet.cpp

```diff
@@ -70,6 +71,47 @@ const CWalletTx* CWallet::GetWalletTx(const uint256& hash) const
     return &(it->second);
 }
 
+// Generate a new spending key and return its public payment address
+CZCPaymentAddress CWallet::GenerateNewZKey()
+{
+    AssertLockHeld(cs_wallet); // mapZKeyMetadata
+    auto k = SpendingKey::random();
+    auto addr = k.address();
+
+    // Check for collision, even though it is unlikely to ever occur
+    if (CCryptoKeyStore::HaveSpendingKey(addr))
+        throw std::runtime_error("CWallet::GenerateNewZKey(): Collision detected");
```

Comment:
On second thought, this does provide some defense-in-depth for future maintenance: suppose key generation is changed in the future in an insecure manner we don't realize, but this check is left in place. Then it provides some protection. Switching from random to deterministic key generation is a perfect example, because that might happen in code elsewhere.

Now I'm just curious how it's tested.